src/main/scala/mesosphere/marathon/upgrade/ScalingProposition.scala

```diff
@@ -0,0 +1,9 @@
+package mesosphere.marathon.upgrade
+
+import mesosphere.marathon.Protos.MarathonTask
+
+class ScalingProposition(private val candidatesToKill: Seq[MarathonTask],
+                         private val numberOfTasksToStart: Int) {
+  val tasksToKill = if (candidatesToKill.nonEmpty) Some(candidatesToKill) else None
```

Comment:
I originally wanted to avoid construction of a ScalingProposition with, e.g.
 ScalingProposition(Some(Seq.empty[...]), Some(-100))
because this is semantical nonsense. The class as written prevents the programmer from doing this. I admit, though, that the implementation is somewhat weird because it looks like a case class but is not.